lib/kernel/src/code_server.erl

```diff
@@ -728,25 +728,17 @@ do_check_path([Dir | Tail], PathChoice, ArchiveExt, Acc) ->
 %%
 %% Add new path(s).
 %%
-add_path(Where,Dir,Path,Cache,NameDb) when is_atom(Dir) ->
-    add_path(Where,atom_to_list(Dir),Path,Cache,NameDb);
 add_path(Where,Dir0,Path,Cache,NameDb) when is_list(Dir0) ->
-    case int_list(Dir0) of
-	true ->
-	    Dir = filename:join([Dir0]), % Normalize
-	    case check_path([Dir]) of
-		{ok, [NewDir]} ->
-		    {true, do_add(Where,NewDir,Path,Cache,NameDb)};
-		Error ->
-		    {Error, Path}
-	    end;
-	false ->
-	    {{error, bad_directory}, Path}
+    Dir = filename:join([Dir0]), % Normalize
```

Comment:
Isn't this already done in the client in `normalize_paths`?

Reply:
Not really:

```
1> filename:join(["foo//bar"]).
"foo/bar"
```

We could move it to normalize paths though and do more work on the client (it is on my to do list), but I am being very conservative with these changes. For this PR, I tried to change the `code_server` the minimum necessary to support the changes to `erl_prim_loader`.

I also want to revise all uses of `filename:*` in `code_server`, because I think there is some duplication too.

Reply:
That definitely makes sense